### PR TITLE
Xl underwear and accessories Part 1

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
SUMMARY: Content "expanded oversized underwear and accesories"

#### Purpose of change
@LaVeyanFiend pointed out that XL underwear and belts don't exist in mainline despite the possibility that mutants can grow to huge size and might want them.  This will likely be a multi part PR adding XL items and their recipes, however at current I will not be adding them to any itemgroups as they likely wouldn't exist pre Cataclysm.

#### Describe the solution
Adds the ability to fully dress for our Huge mutant friends.

#### Describe alternatives you've considered
none

#### Testing
tbd

### CODE REVIEW

File: data/json/items/armor/undergarment.json [hunk 1]
```diff
@@ -169,6 +188,25 @@
     "material_thickness": 1,
     "flags": [ "VARSIZE", "SKINTIGHT" ]
   },
+  {
+    "id": "xLboy_shorts",
```
Comment (LaVeyanFiend) [inline data/json/items/armor/undergarment.json:192]:
```suggestion
    "id": "xlboy_shorts",
```

File: data/json/items/armor/boots.json [hunk 2]
```diff
@@ -1264,6 +1264,25 @@
     "material_thickness": 1,
     "flags": [ "VARSIZE", "SKINTIGHT" ]
   },
+  {
+    "id": "xlsocks",
+    "type": "ARMOR",
+    "name": { "str_sp": "pair of xl socks" },
```
Comment (LaVeyanFiend) [inline data/json/items/armor/boots.json:1270]: The other XL items have theirs capitalized
```suggestion
    "name": { "str": "pair of XL socks", "str_pl": "pairs of XL socks" },
```

File: data/json/items/armor/boots.json [hunk 3]
```diff
@@ -1264,6 +1264,25 @@
...
+    "description": "Socks.  Big ones.  Put 'em on your feet.",
+    "weight": "58 g",
+    "volume": "250 ml",
+    "price": 200,
+    "price_postapoc": 50,
+    "material": [ "cotton" ],
+    "symbol": "[",
+    "looks_like": "socks_wool",
```
Comment (LaVeyanFiend) [inline data/json/items/armor/boots.json:1278]:
```suggestion
    "looks_like": "socks",
```
So the looks_like hierarchy looks for the original socks' texture before looking for the wool socks, if a tileset has a seperate image for regular ones.

File: data/json/items/armor/undergarment.json [hunk 4]
```diff
@@ -150,6 +150,25 @@
     "material_thickness": 1,
     "flags": [ "VARSIZE", "SKINTIGHT" ]
   },
+  {
+    "id": "xlboxer_shorts",
+    "type": "ARMOR",
+    "name": { "str_sp": "xl boxer shorts" },
```
Comment (LaVeyanFiend) [inline data/json/items/armor/undergarment.json:156]:
```suggestion
    "name": { "str_sp": "XL boxer shorts" },
```

File: data/json/items/armor/undergarment.json [hunk 5]
```diff
@@ -169,6 +188,25 @@
     "material_thickness": 1,
     "flags": [ "VARSIZE", "SKINTIGHT" ]
   },
+  {
+    "id": "xLboy_shorts",
+    "type": "ARMOR",
+    "name": { "str_sp": "xl boy shorts" },
```
Comment (LaVeyanFiend) [inline data/json/items/armor/undergarment.json:194]:
```suggestion
    "name": { "str_sp": "XL boy shorts" },
```

File: data/json/items/armor/undergarment.json [hunk 6]
```diff
@@ -503,6 +541,25 @@
     "material_thickness": 1,
     "flags": [ "VARSIZE", "WATER_FRIENDLY", "SKINTIGHT" ]
   },
+  {
+    "id": "xlsports_bra",
+    "type": "ARMOR",
+    "name": { "str": "xl sports bra" },
```
Comment (LaVeyanFiend) [inline data/json/items/armor/undergarment.json:547]:
```suggestion
    "name": { "str": "XL sports bra" },
```

**Comment by Maleclypse** [comment]
> I'd also suggest adding XL wool & flame-resistant socks

Anything else you think I should put in before I move on to the AFS only portion for things like `police_belt` and other pre-Cataclysm gear that only makes sense in a world with `HUGE` individuals?

**Comment by LaVeyanFiend** [comment]
I had a thought on #39210 adding an "XL-ify"(name pending :P) option to save a lot of work defining seperate XL versions of each item.

**Pull request merged**